### PR TITLE
feat: order details page initialization

### PR DESCRIPTION
**Pull request by irina-sid**
The type of this PR is: **feature**

This PR resolves [PX-4070]

### Description

Order Details page is registered in the route registry and has a relay container. So user can click on View Order button on Orders history screen and see the Order Details page with rendered OrderId on it.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

iPhone 11 Pro

[PX-4070]: https://artsyproduct.atlassian.net/browse/PX-4070

### CODE REVIEW

File: CHANGELOG.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ upcoming:
     - Add order history to user profile (behind feature flag) - alexj105
     - Add order history skeleton (behind feature flag) - Serge0n
     - Add orders to order history (behind feature flag) - Serge0n
+    - Add order details page (app registry, a route with orderID, relay container)
```
Comment (brainbicycle) [inline CHANGELOG.yml:23]:
```suggestion
    - Add order details page (app registry, a route with orderID, relay container) - irina-sid
```
minor suggestion to add username just so we know who to reach out to about the change

**Review by brainbicycle — APPROVED** [review]
LGTM!

File: src/lib/navigation/routes.tsx [hunk 2]
```diff
@@ -211,6 +211,7 @@ function getDomainMap(): Record<string, RouteMatcher[] | null> {
 
     new RouteMatcher("/city-bmw-list/:citySlug", "CityBMWList"),
     new RouteMatcher("/make-offer/:artworkID", "MakeOfferModal"),
+    new RouteMatcher("/order-history/:orderID", "OrderDetails"),
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/lib/navigation/routes.tsx:214]: **issue:(blocker)** On android to get the deep linking for this page to work properly,we will need to add the route to  `AndroidManifest.xml`

**Comment by MounirDhahri** [comment]
This looks good 🙌 I left few comments @irina-sid that would be great to be addressed in an other PR before we release

File: CHANGELOG.yml (hunk 1 above)
Comment (irina-sid) [reply CHANGELOG.yml:23]: Sure!

Comment (irina-sid) [reply CHANGELOG.yml:23]: Sure!

**Pull request merged**